### PR TITLE
CNX-9877-units-fix

### PR DESCRIPTION
**Pull request by bjoernsteinhagen**
## Description & motivation
Two community issues related to the SAP 2000 Connector doing nothing (both **Send** and **Receive**). For more context, the community issues are linked here:

- https://speckle.community/t/why-receiving-and-sending-model-dont-work-on-sap2000/12245/4
- https://speckle.community/t/question-is-it-possible-for-me-to-use-the-sap2000-connector-with-v25-2-0/12032

When reviewing the Report, the following failed message was noted: `Unable to retrieve valid length units from the ETABS document`, giving an indication of the error source.
## Changes:
- Re-introducing the old method for obtaining model units, this change happened in #3410. The 'old method' is needed for SAP 2000 and CSIBridge (`GetPresentUnits()`).
- The change from the aforementioned PR remains relevant for ETABS and SAFE (`GetPresentUnits_2()`)
- A function for each respective method was created:
  - `GetModelUnitsFromETABS()`
  - `GetModelUnitsFromSAP()`
- Links to the appropriate call depending on the calling application with preprocessor directives.

## Screenshots:

- From the community forum, model received and noted as "successfully completed", however, SAP model is **empty**:

image

- After above mentioned fixes, model is well received in SAP 2000:

image

## Validation of changes:

- Fixes to SAP 2000 can have adverse effects on the other CSI products as such, sending and receiving in SAP 2000 and ETABS respectively was tested. 

## Checklist:
- [x] My pull request follows the guidelines in the [Contributing guide](https://github.com/specklesystems/speckle-server/blob/main/CONTRIBUTING.md)?
- [x] My pull request does not duplicate any other open [Pull Requests](../../pulls) for the same update/change?
- [x] My commits are related to the pull request and do not amend unrelated code or documentation.
- [x] My code follows a similar style to existing code.
- [ ] I have added appropriate tests.
- [ ] I have updated or added relevant documentation.

### CODE REVIEW

**Review by connorivy — APPROVED** [review]
Tested and looks good 👍

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs [hunk 1]
```diff
@@ -27,18 +27,45 @@ public string ModelUnits()
...
+    return lengthUnits.ToString();
+  }
+
+  public string GetModelUnitsFromSAP()
+  {
+    // GetPresentUnits() works for SAP 2000 and CSIBridge
+    var units = Model.GetPresentUnits();
+    if (units != 0)
+    {
+      string[] unitsCat = units.ToString().Split('_');
```
Comment (JR-Morgan) [inline Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:65]: Are we sure its safe to assume that `GetPresentUnits` will always return a value that has an `'_'`?

Is this documented in their API?

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs [hunk 2]
```diff
@@ -27,18 +27,45 @@ public string ModelUnits()
       return _modelUnits;
     }
 
+#if ETABS || SAFE
+
+    _modelUnits = GetModelUnitsFromETABS();
+
+#else
```
Comment (JR-Morgan) [inline Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:34]: What about CSI Bridge?

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs (hunk 1 above)
Comment (bjoernsteinhagen) [reply Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:65]: From the SAP 2000 documentation:

image

Hence index positions for 0, 1 and 2 being force, distance and temp respectively. In SAP interface these enums seem well enforced (I, as the user, can't define something funny):

image

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs (hunk 2 above)
Comment (bjoernsteinhagen) [reply Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:34]: It's my understanding that SAP 2000 and CSiBridge are the same.:

image

But I have requested a trial for CSiBrdige to get my hands on the documentation to confirm. Will keep you in the loop

Comment (JR-Morgan) [reply Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:34]: Ok that sounds good. I'm happy to merge without testing, but if you could check with @connorivy how we can test CSiBridge.
I'd like to do a quick test before we make the proper stable 2.20 release. But I'm not sure what licencing we have etc..

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs (hunk 1 above)
Comment (JR-Morgan) [reply Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:65]: Looks good

**Review by JR-Morgan — APPROVED** [review]

File: Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs (hunk 1 above)
Comment (bjoernsteinhagen) [reply Objects/Converters/ConverterCSI/ConverterCSIShared/ConverterCSIUtils.cs:65]: Just got my hands on the documentation, identical to SAP 2000 😄

**Pull request merged**